### PR TITLE
Adds delays for suggestion api (uplift to 1.3)

### PR DESCRIPTION
**Pull request by bsclifton**
A more concise manual merge resolution of https://github.com/brave/brave-core/pull/4441 

Uplift of https://github.com/brave/brave-core/pull/4372
Fixes https://github.com/brave/brave-browser/issues/7709
Fixes https://github.com/brave/brave-browser/issues/6940

Approved, please ensure that before merging: 
- [ ] You have checked CI and the builds, lint, and tests all pass or are not related to your PR. 
- [x] You have tested your change on Nightly. 
- [x] The PR milestones match the branch they are landing to. 

After you merge: 
- [ ] The associated issue milestone is set to the smallest version that the changes is landed on.

### CODE REVIEW

**Comment by bsclifton** [comment]
@tmancey can you help me check this out? I think we're good to go here... it builds, unit tests all pass. I only pulled in the minimum required

Once we verify it, we can assign to @brave/uplift-approvers for consideration 😄

**Comment by tmancey** [comment]
@bsclifton LGTM++

**Comment by bsclifton** [comment]
Might still be some additional work. Build worked fine, unit tests all pass. But looks like browser tests have a failure:
https://ci.brave.com/job/brave-browser-build-pr/job/suggestion-api-1.3_2nd_try/1/execution/node/596/log/
> *2 tests failed*:
>     BraveRewardsBrowserTest.AutoContribution (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:2189)
>     BraveRewardsBrowserTest.ProcessPendingContributions (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:2717)
> *6 tests crashed*:
>     BraveRewardsBrowserTest.MultipleRecurringOverBudgetAndPartialAutoContribution (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:3132)
>     BraveRewardsBrowserTest.RecurringAndPartialAutoContribution (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:3085)
>     BraveRewardsBrowserTest.RecurringTipForVerifiedPublisher (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:2313)
>     BraveRewardsBrowserTest.TipConnectedPublisherAnon (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:2940)
>     BraveRewardsBrowserTest.TipConnectedPublisherAnonAndConnected (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:2960)
>     BraveRewardsBrowserTest.TipVerifiedPublisher (../../brave/components/brave_rewards/browser/rewards_service_browsertest.cc:2274)

File: vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom [hunk 1]
```diff
@@ -1,13 +1,24 @@
+// Copyright (c) 2019 The Brave Authors. All rights reserved.
 // This Source Code Form is subject to the terms of the Mozilla Public
-// License, v. 2.0. If a copy of the MPL was not distributed with this file,
-// You can obtain one at http://mozilla.org/MPL/2.0/.
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
 module ledger.mojom;
 
+enum ContributionStep {
+  STEP_AC_TABLE_EMPTY = -4,
```
Comment (masparrow) [inline vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom]: Can I ask why start steps at `-4`?

Comment (bsclifton) [reply vendor/bat-native-ledger/include/bat/ledger/public/interfaces/ledger.mojom]: I have no idea- the original PR does this (this is just an uplift)
https://github.com/brave/brave-core/pull/4372

Maybe @tmancey can provide more context?

**Comment by bsclifton** [comment]
OK problem resolved- force pushed change after verifying all tests pass locally! 🎉

**Comment by kjozwiak** [comment]
Looks like `test-install` failed on `macOS` as per https://ci.brave.com/job/brave-browser-build-pr/job/suggestion-api-1.3_2nd_try/2/execution/node/698/log/ which is a known issue that's not related to this specific PR. All the other platforms passed without any errors/failures.

**Pull request merged**